packages/loader/container-loader/src/deltaManager.ts

```diff
@@ -828,7 +843,7 @@ export class DeltaManager
                 // detected gap, this gap can't be filled in later on through websocket).
                 // And in practice that does look like the case. The place where this code gets hit is if we lost
                 // connection and reconnected (likely to another box), and new socket's initial ops contains these ops.
-                if (op.sequenceNumber >= to) {
+                if (op.sequenceNumber >= lastExpectedOp) {
```

Comment:
Is the `push` event for this listener guaranteed to be listening to sequential ops? I'm wondering if there is a case where it could abort the storage call even though it shouldn't have.

It might be safer if this also checked against `this.lastProcessedSequenceNumber`. The listener could be changed to the `DeltaManager` `op` event instead (because that is emitted after `lastProcessedSequenceNumber` is moved forward).

Reply:
Event is on this._inbound queue. ops are pushed into that queue always sequentially, i.e. ops already went through all the logic that ensures dedupping, filling in gaps, etc.
So it's fine here.

The place that is a bit icky is actually code above that I've added that tests this.lastProcessedSequenceNumber. It's equivalent to above if op processing is going through. However, if op processing is paused for any reason, then this._inbound would be getting ops pushed to it and would raise "push" event, but it would not dispose ops, and thus  this.lastProcessedSequenceNumber would not be updated. Thus changing condition to test this.lastQueuedSequenceNumber as that number is same as this.lastProcessedSequenceNumber + this._inbound.length and represents last sequential op that we saw.

I'll add assert here as well to ensure they are the same things